src/lib/CPAuth/CPAuth.ts

```diff
@@ -0,0 +1,36 @@
+import { getAuthorityURLFromGSApiURL } from 'lib/CPAuth/util';
+import OAuth2, { IOAuth2Config } from 'lib/OAuth2/OAuth2';
+import { AppRoutes } from 'shared/constants/routes';
+
+const defaultConfig: IOAuth2Config = {
+  authority: getAuthorityURLFromGSApiURL(window.config.apiEndpoint),
+  clientId: 'zQiFLUnrTFQwrybYzeY53hWWfhOKWRAU',
+  clientSecret: 'TVHzVPin2WTiCma6bqp5hdxgKbWZKRbz',
+  redirectUri: `${window.location.origin}${AppRoutes.CPAccessCallback}`,
+  scope:
+    'openid offline_access profile email groups audience:server:client_id:dex-k8s-authenticator',
+  responseType: 'code',
+  responseMode: 'query',
+  prompt: 'none',
+  automaticSilentRenew: true,
+  includeIDTokenInSilentRenew: true,
+  loadUserInfo: true,
+  revokeAccessTokenOnLogout: true,
+  filterProtocolClaims: true,
+  validateSubOnSilentRenew: true,
+  persistenceMethod: localStorage,
+};
+
+class CPAuth extends OAuth2 {
```

Comment:
This inherits from the `OAuth2` class, so we could have this as a singleton, but not limit the number of OAuth2 providers to a single one.